overlord/snapstate/snapstate.go

```diff
@@ -203,6 +203,13 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	addTask(setupAliases)
 	prev = setupAliases
 
+	// on refresh re-connect existing connections and run their interface hooks (if applicable)
+	if snapst.IsInstalled() {
+		reconnectTask := st.NewTask("reconnect", fmt.Sprintf(i18n.G("Re-connect the interfaces and re-run interface hooks for snap %q"), snapsup.Name()))
```

Comment:
"Reconnect interfaces for snap %q".. hooks are part of interfaces (note s/the//).

Reply:
Done.